go/app/server.go

```diff
@@ -151,15 +176,90 @@ func (s *Handlers) AddItem(w http.ResponseWriter, r *http.Request) {
 // this method calculates the hash sum of the image as a file name to avoid the duplication of a same file
 // and stores it in the image directory.
 func (s *Handlers) storeImage(image []byte) (filePath string, err error) {
-	// STEP 4-4: add an implementation to store an image
-	// TODO:
-	// - calc hash sum
-	// - build image file path
-	// - check if the image already exists
-	// - store image
-	// - return the image file path
+	// 1. ハッシュを計算（SHA-256）
+	hash := sha256.Sum256(image)
+	hashString := hex.EncodeToString(hash[:]) // 16進数の文字列に変換
+
+	fileName := filepath.Join(hashString + ".jpg")
+	if err := os.MkdirAll(s.imgDirPath, os.ModePerm); err != nil {
+		return "", err
+	}
+
+	// 2. 画像の保存先パスを生成
+	filePath = filepath.Join(s.imgDirPath, fileName)
 
-	return
+	// 3. 同じパス(ハッシュ)の画像がすでに存在するのかをチェックする
+	if _, err := os.Stat(filePath); err == nil {
+		return filePath, nil
+	}
+
+	// 4. 画像を保存
+	if err := StoreImage(filePath, image); err != nil {
+		return "", err
+	}
+
+	// 5. 保存したファイルのパスを返す
+	return filePath, nil
```

Comment:
Sorry just one comment, you might have missed it, but please return `fileName` here instead of `filePath`
```suggestion
	return fileName, nil
```

Reply:
@pandakun26 
I filled in PR description! Could you check it again?